proto/aserto/directory/reader/v3/reader.proto

```diff
@@ -0,0 +1,359 @@
+syntax = "proto3";
+
+package aserto.directory.reader.v3;
+
+option go_package = "github.com/aserto-dev/go-directory/aserto/directory/reader/v3;reader";
+
+import "google/api/annotations.proto";
+import "protoc-gen-openapiv2/options/annotations.proto";
+
+import "aserto/directory/common/v3/common.proto";
+
+service Reader {
+    // object type metadata methods
+    rpc GetObjectType(GetObjectTypeRequest) returns (GetObjectTypeResponse) {};
+    rpc GetObjectTypes(GetObjectTypesRequest) returns (GetObjectTypesResponse) {};
+
+    // relation type metadata methods
+    rpc GetRelationType(GetRelationTypeRequest) returns (GetRelationTypeResponse) {};
+    rpc GetRelationTypes(GetRelationTypesRequest) returns (GetRelationTypesResponse) {};
+
+    // permission metadata methods
+    rpc GetPermission(GetPermissionRequest) returns (GetPermissionResponse) {};
+    rpc GetPermissions(GetPermissionsRequest) returns (GetPermissionsResponse) {};
+
+    // object methods
+    rpc GetObject(GetObjectRequest) returns (GetObjectResponse) {
+        option (google.api.http) = {
+            get: "/api/v3/directory/object/{object_type}/{object_id}"
+        };
+        option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+            tags: "directory"
+            summary: "Get object instance"
+            description: "Returns single object instance, optionally with relations."
+            operation_id: "directory.v3.object.get"
+            deprecated: false
+            security: {
+                security_requirement: {
+                    key: "TenantID";
+                    value: {}
+                }
+                security_requirement: {
+                    key: "DirectoryAPIKey";
+                    value: {}
+                }
+            }
+        };
+    };
+
+    rpc GetObjectMany(GetObjectManyRequest) returns (GetObjectManyResponse) {};
+    rpc GetObjects(GetObjectsRequest) returns (GetObjectsResponse) {
+        option (google.api.http) = {
+            get: "/api/v3/directory/objects/{object_type}"
```

Comment:
Does this definition allow GET requests to `/.../directory/objects/` to get objects of all types? The generated OpenAPI spec shows the `object_type` parameter as required (https://github.com/aserto-dev/openapi-directory/pull/1/files#diff-0fd0bfb20083d99259c5c2616b67772c24b79e5390527f56112a7c4e4a0a2aa2R212)

Reply:
That is an OpenAPI issue, that when you have a string instead of a *string it default to required. 
This will get fixed adding explicit OpenAPI annotations